### PR TITLE
uint32_to_string => str_sprintf

### PR DESCRIPTION
**Pull request by MarioGK**
## Description:
Updating documentation related to the RDM6300 example.

### CODE REVIEW

File: components/binary_sensor/rdm6300.rst [hunk 1]
```diff
@@ -68,7 +68,7 @@ unsigned integer.
         then:
           - mqtt.publish:
               topic: rdm6300/tag
-              payload: !lambda 'return uint32_to_string(x);'
+              payload: !lambda 'return str_sprintf("%u", x);'
```
Comment (OttoWinter) [inline components/binary_sensor/rdm6300.rst:71]:
```suggestion
              payload: !lambda 'return to_string(x);'
```

File: components/binary_sensor/rdm6300.rst [hunk 2]
```diff
@@ -79,7 +79,7 @@ using :ref:`api-homeassistant_tag_scanned_action`.
       # ...
       on_tag:
         then:
-          - homeassistant.tag_scanned: !lambda 'return uint32_to_string(x);'
+          - homeassistant.tag_scanned: !lambda 'return str_sprintf("%u", x);'
```
Comment (OttoWinter) [inline components/binary_sensor/rdm6300.rst:82]:
```suggestion
          - homeassistant.tag_scanned: !lambda 'return to_string(x);'
```

More readable IMO

Comment (MarioGK) [reply components/binary_sensor/rdm6300.rst:82]: Looks good

Comment (OttoWinter) [reply components/binary_sensor/rdm6300.rst:82]: Hi @MarioGK, could you press the apply button on these suggestions? The PR still has the `str_sprintf` in it.

**Comment by flaviut** [comment]
Manually merged in https://github.com/esphome/esphome-docs/pull/1956

**Pull request closed without merging**